server/api/router/series.ts

```diff
@@ -0,0 +1,129 @@
+                columns: {
+                  id: true,
+                  seriesId: true,
+                  userId: true
+                },
+                with: {
+                  series: {
+                    columns: {
+                      id: true,
+                      title: true
+                    },
+                  },
+                },
+                where: (post, { eq }) => eq(post.id, postId),
+              });
+
+            if (!currentPost) {
+                throw new TRPCError({ code: 'NOT_FOUND' });
+            }
+            if (currentPost?.userId !== ctx.session.user.id) {
+                throw new TRPCError({
+                  code: "FORBIDDEN",
+                });
+            }
+            const createNewSeries = async (seriesTitle: string) => {
+              // check if a series with that name already exists
+              // or else create a new one
+              return await ctx.db.transaction(async (tx) => {
+                let seriesId : number;
+                const currSeries = await tx.query.series.findFirst({
+                    columns: {
+                    id: true
+                    },
+                    where: (series, { eq }) => eq(series.title, seriesTitle),
+                })
+                
+                if(!currSeries){
+                    const [newSeries] = await tx.insert(series).values({
+                    title: seriesTitle, 
+                    userId: ctx.session.user.id,
+                    updatedAt: new Date()  
+                    }).returning();
+                    
+                    seriesId = newSeries.id;
+                }
+                else{
+                    seriesId = currSeries.id;
+                }
+                    // update that series id in the current post
+                    await tx
+                    .update(post)
+                    .set({
+                        seriesId: seriesId
+                    })
+                    .where(eq(post.id, currentPost.id));
+                })
+
+            }
+
+            const unlinkSeries = async (seriesId: number) => {
+              // Check if the user has added a another post with the same series id previously
+                return await ctx.db.transaction(async (tx) =>{
+                    const anotherPostInThisSeries = await tx.query.post.findFirst({
+                        where: (post, { eq, and, ne }) => 
+                            and (
+                            ne(post.id, currentPost.id),
+                            eq(post.seriesId, currentPost.seriesId!)
+                            )
+                    })
+                    // if another post with the same seriesId is present, then do nothing
+                    // else remove the series from the series table
+                    if(!anotherPostInThisSeries){
+                        await tx.delete(series).where(eq(series.id, seriesId));
+                    }
```

Comment:
Since the seriesId is being picked from the currentPost <Which will raise an error if the unauthorized user accesses>, so this comparison is not necessary in my opinion. Had the comparison been done based on series title, then comparing the user id would have been important.

However, this check is alse added in the latest commit.